gradle-sls-packaging/src/main/groovy/com/palantir/gradle/dist/tasks/ConfigTarTask.groovy

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2016 Palantir Technologies
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * <http://www.apache.org/licenses/LICENSE-2.0>
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.dist.tasks
+
+import com.palantir.gradle.dist.service.JavaServiceDistributionPlugin
+import org.gradle.api.Project
+import org.gradle.api.tasks.bundling.Compression
+import org.gradle.api.tasks.bundling.Tar
+
+class ConfigTarTask {
+    static Tar createConfigTarTask(Project project, String taskName, String productType) {
+        project.tasks.create(taskName, Tar) { p ->
+            p.group = JavaServiceDistributionPlugin.GROUP_NAME
+            p.description = "Creates a compressed, gzipped tar file that contains the sls configuration files for the product"
+            // Set compression in constructor so that task output has the right name from the start.
+            p.compression = Compression.GZIP
+            // The extension is the product type without the version
+            // service.v1 -> .service.config.tgz
```

Comment:
should assert that the type has this format and fail if it doesn't

Reply:
I'm fine adding a check for this, but it looks like the distribution extension already enforces that it comes from a predefined list of types: 
```java
void productType(String type) {
    if (!VALID_PRODUCT_TYPES.contains(type)) {
        throw new IllegalArgumentException("Invalid product type '${type}' specified; supported types: ${VALID_PRODUCT_TYPES}.")
    }
    this.productType = type
}
```

Reply:
kind of brittle, but ok